v2.3.0 gives internal nil pointer on "phase.RLock()" but 2.2.1 does work.

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### Naabu version: GO Library 2.3.0 (See packages used below)
<!-- You can find current version of naabu with "naabu -version" -->
<!-- We only accept issues that are reproducible on the latest version of naabu. -->
<!-- You can find the latest version of project at https://github.com/projectdiscovery/naabu/releases/ -->
This is using the go library, and two packages being imported are

```
	"github.com/projectdiscovery/naabu/v2/pkg/result"
	"github.com/projectdiscovery/naabu/v2/pkg/runner"
```

These are all set to 2.3.0 (the latest) at the time of writing.


### Current Behavior:
I am using Nuclei and Naabu, but they have entirely separate functions. If I run a nuclei scan then a nil pointer and a stack trace is coming from "naabu", I will paste this here :-

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102fe4e64]

goroutine 52 [running]:
github.com/projectdiscovery/naabu/v2/pkg/scan.(*Phase).Is(0x14001b86c01?, 0x2?)
        /Users/ian/.gvm/pkgsets/go1.22/global/pkg/mod/github.com/projectdiscovery/naabu/v2@v2.3.0/pkg/scan/scan.go:50 +0x24
github.com/projectdiscovery/naabu/v2/pkg/scan.TransportReadWorker.func1({{{0x14008af32e2, 0x2c, 0x2c}, {0x14008af32e2, 0x0, 0x0}}, 0x50, 0xcd2e, 0xc152c4e9, 0x9d744071, ...}, ...)
        /Users/ian/.gvm/pkgsets/go1.22/global/pkg/mod/github.com/projectdiscovery/naabu/v2@v2.3.0/pkg/scan/scan_unix.go:528 +0xbc
github.com/projectdiscovery/naabu/v2/pkg/scan.TransportReadWorker.func3(0x14000b986c0)
        /Users/ian/.gvm/pkgsets/go1.22/global/pkg/mod/github.com/projectdiscovery/naabu/v2@v2.3.0/pkg/scan/scan_unix.go:641 +0x55c
created by github.com/projectdiscovery/naabu/v2/pkg/scan.TransportReadWorker in goroutine 35
        /Users/ian/.gvm/pkgsets/go1.22/global/pkg/mod/github.com/projectdiscovery/naabu/v2@v2.3.0/pkg/scan/scan_unix.go:599 +0x140
task: Failed to run task "module": exit status 2
```

It's worth mentioning that this is NOT running any NAABU code. Only the "imports" have been executed because they are in a "discovery" struct that I set up—but no code is being executed. I believe that some NAABU code is executed on valid imports because it uses a number of "init" functions.

I tried debugging and noticed that "phase" is nil in this struct / method in the naabu source code.

```
func (phase *Phase) Is(state State) bool {
	phase.RLock() ///// <<<< This is where the issue is. phase is nil.
	defer phase.RUnlock()

	return phase.State == state
}
```

From what I understand, nuclei does not depend on any naabu code.

Rolling naabu back to 2.2.1 fixes this issues.

The current test machine is a Mac M1.


### Expected Behavior:
There should be no internal panic on v2.3.0

--- a/v2/pkg/scan/scan_common.go
+++ b/v2/pkg/scan/scan_common.go
@@ -38,10 +38,14 @@ type ListenHandler struct {
 	TcpChan, UdpChan, HostDiscoveryChan    chan *PkgResult
 }
 
+func NewListenHandler() *ListenHandler {
+	return &ListenHandler{Phase: &Phase{}}
+}
+
 func Acquire(options *Options) (*ListenHandler, error) {
 	// always grant to unprivileged scans or connect scan
 	if PkgRouter == nil || !privileges.IsPrivileged || options.ScanType == "c" {
-		return &ListenHandler{Phase: &Phase{}}, nil
+		return NewListenHandler(), nil
 	}
 
 	for _, listenHandler := range ListenHandlers {

--- a/v2/pkg/scan/scan_unix.go
+++ b/v2/pkg/scan/scan_unix.go
@@ -61,7 +61,7 @@ func init() {
 
 	// pre-reserve up to 10 ports
 	for i := 0; i < NumberOfHandlers; i++ {
-		var listenHandler ListenHandler
+		listenHandler := NewListenHandler()
 		if port, err := freeport.GetFreeTCPPort(""); err != nil {
 			gologger.Error().Msgf("could not setup get free port: %s", err)
 			return

--- a/v2/pkg/scan/scan_unix.go
+++ b/v2/pkg/scan/scan_unix.go
@@ -102,7 +102,7 @@ func init() {
 		go listenHandler.UdpReadWorker4()
 		go listenHandler.UdpReadWorker6()
 
-		ListenHandlers = append(ListenHandlers, &listenHandler)
+		ListenHandlers = append(ListenHandlers, listenHandler)
 	}
 
 	handlers = &Handlers{}

